cmd/frontend/graphqlbackend/site.go

```diff
@@ -548,10 +549,11 @@ func (r *siteResolver) RequiresVerifiedEmailForCody(ctx context.Context) bool {
 	// App is typically forwarding Cody requests to dotcom.
 	// This section can be removed if dotcom stops requiring verified emails
 	if deploy.IsApp() {
-		// App users can specify their own keys
-		// if they use their own keys requests are not forwarded to dotcom
-		// which means a verified email is not needed
-		return conf.Get().Completions == nil
+		c := conf.GetCompletionsConfig(conf.Get().SiteConfig())
+		// App users can specify their own keys using one of the regular providers.
+		// If they use their own keys requests are not going through Cody Gateway
+		// which means a verified email is not needed.
+		return c != nil && c.Provider != conftypes.CompletionsProviderNameSourcegraph
```

Comment:
```suggestion
		return c == nil || c.Provider == conftypes.CompletionsProviderNameSourcegraph
```
It think this is backwards, App should require a verified email if it's using the default empty config or if the provider is `sourcegraph`  The dotcom user actor in the gateway checks if the user had verified emails and will return not enabled if there are no verified emails.

Reply:
thanks, fixed!